presto-orc/src/main/java/com/facebook/presto/orc/ColumnGroupReader.java

```diff
@@ -644,9 +589,7 @@ private void alignResultsAndRemoveFromQualifyingSet(int numAdded, int lastStream
             }
             QualifyingSet output = reader.getOutputQualifyingSet();
             QualifyingSet input = reader.getInputQualifyingSet();
-            int truncationRow = reader.getTruncationRow();
-            if (truncationRow == -1 &&
-                    (output == null || output.getPositionCount() == input.getPositionCount() && !output.hasErrors())) {
+            if (output == null || output.getPositionCount() == input.getPositionCount() && !output.hasErrors()) {
```

Comment:
This statement is confusing. It says if (a || b && c). && has precedence over ||, but b and c use output which can be null and therefore may result in NPE.